userspace/libsinsp/threadinfo.h

```diff
@@ -354,22 +354,28 @@ class SINSP_PUBLIC sinsp_threadinfo
 protected:
 	inline sinsp_fdtable* get_fd_table()
 	{
-		sinsp_threadinfo* root;
-
 		if(!(m_flags & PPM_CL_CLONE_FILES))
 		{
-			root = this;
+			return &m_fdtable;;
 		}
 		else
 		{
-			root = get_main_thread();
-			if(NULL == root)
-			{
-				return NULL;
-			}
+			sinsp_threadinfo* root = get_main_thread();
+			return (root == nullptr) ? nullptr : &(root->m_fdtable);
 		}
+	}
 
-		return &(root->m_fdtable);
+	inline const sinsp_fdtable* get_immutable_fd_table() const
```

Comment:
Make the immutable version the default?

Reply:
I did in my original change, but wanted to limit the magnitude of the change

Reply:
Understood. I'd still prefer `get_fd_table() const` + `get_fd_table_mut()` (or whatever) but your call.